recipes/mastercard/runner.sh

```diff
@@ -34,20 +47,25 @@ VERSION=$DATE
         exit 3
     fi
 
-    #will download all files from mastercard (there has already been a check to verify there is one and only one zip file). Then mastercard will delete after successfull download.
-    scp -P 22022  -i /root/.ssh/id_rsa_axway -o "StrictHostKeyChecking=no" newyorkcity@files.mastercard.com:geoinsights/data/fromMC/* /input
+    #will download all files from mastercard. Then mastercard will delete after successfull download. May be more than one file without check.
+    echo 'downloading from mastercard'
+    scp -P 22022 -i /root/.ssh/id_rsa_axway -o "StrictHostKeyChecking=no" newyorkcity@files.mastercard.com:geoinsights/data/fromMC/* ./input
     #For testing purposes
-    #cp Geogrids_NYC_Zip_Code_Level_Jan2019_Feb2021.zip input/
+    #cp Geogrids_NYC_Zip_Codes_Level_01Jan2019_25Apr2021_Final.zip input/
 
-    #upload files to aws: 
-    aws s3 cp ./input/ s3://recovery-data-partnership/mastercard/ --recursive
+    #upload files to aws for backup. Can handle multiple files.: 
+    #getting InvalidAccessKeyIDError. Commented out until resolved.
+    echo 'uploading to RDP AWS S3'
+    aws s3 cp ./input/ s3://recovery-data-partnership/mastercard/ --recursive || AWS_ERROR=1
 
     #verify the correct file
-    echo "unzipping " $( ls -ltr input | tail -1 | awk '{print $NF}') 
+    echo 'listing...\n'
+    ls -ltr input
+    FULL_FILENAME=$( ls -ltr input | tail -1 | awk '{print $NF}') 
+    echo "unzipping " $FULL_FILENAME
     #unzips the first file by chronological order by sorting by modified date, reversed, and taking the tail to avoid the header
-    # (there is a check above to ensure there is only be one zip file so sorting is redundant).
     # Then use awk to select filename.
-    unzip -d /input -P $MASTERCARD_PASSWORD $( ls -ltr input | tail -1 | awk '{print $NF}') 
+    unzip -d /input -P $MASTERCARD_PASSWORD $FULL_FILENAME
```

Comment:
where is $MASTERCARD_PASSWORD created? Is it in the .env file?

Reply:
Ah yes. I need to add this to the environment.